pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -126,6 +127,39 @@ type BMCDetails struct {
 	CredentialsName string `json:"credentialsName"`
 }
 
+// RAIDVolume defines the desired configuration of volume in RAID
+type RAIDVolume struct {
+	// Size (Integer) of the logical disk to be created in GiB.
+	// If unspecified or set be 0, the maximum capacity of disk will be used for logical disk.
+	SizeGb int `json:"sizeGb,omitempty"`
+
+	// RAID level for the logical disk. The following levels are supported: 0, 1, 1+0.
+	// +kubebuilder:validation:Enum=0,1,1+0
+	Level nodes.RAIDLevel `json:"level" required:"true"`
+
+	// Name of the volume. Should be unique within the Node. If not specified, volume name will be auto-generated.
```

Comment:
What happens if it's not unique?

Reply:
Should the baremetal-operator do some sort of validation?

Reply:
I would think that if a host was configured with 2 volumes with the same name then something somewhere would complain, even if it's part way through the process of creating the volumes or even the host failing to boot properly because it can't figure out which of the 2 volumes to mount. So it would be good to understand what sort of error behavior is likely to be seen so we can decide how much checking we need to do in metal3 to provide a good experience for users.